.github/workflows/prbuild.yaml

```diff
@@ -0,0 +1,24 @@
+name: Build and Test Pull Request
```

Comment:
I'd call this just "Build and Test" since it runs on push as well (I have a TODO in my head to change this in contour too).